framework/src/systems/NonlinearSystemBase.C

```diff
@@ -3897,17 +3897,19 @@ NonlinearSystemBase::assembleScalingVector()
        as_range(lm_mesh.active_local_elements_begin(), lm_mesh.active_local_elements_end()))
     for (const auto * const field_var : field_variables)
     {
-      mooseAssert(field_var->count() == 1,
-                  "Contact Alex Lindsay and tell him to make this work for array variables.");
-      dof_map.dof_indices(elem, dof_indices, field_var->number());
-      for (const auto dof : dof_indices)
-        scaling_vector.set(dof, field_var->scalingFactor());
+      const auto & factors = field_var->arrayScalingFactor();
+      for (unsigned int i = 0; i < field_var->count(); ++i)
```

Comment:
```suggestion
      for (const auto i : make_range(field_var->count()))
```

Reply:
`count()` is returning unsigned int, so this is consistent. But I can make the change if you insist.

Reply:
I like the addition of being able to specify the index as `const`

Reply:
Fair enough. And also there is only one appearance of `i`.